### PR TITLE
Fix missing null_count() comparison in test framework and related failures

### LINKED ISSUE #6733: [BUG] cudf::apply_boolean_mask returns columns with incorrect null_count
**Describe the bug**
`cudf::apply_boolean_mask` returns columns with incorrect null_count

**Steps/Code to reproduce bug**
Run this unit test
```c++
TEST_F(ApplyBooleanMask, bug)
{
  cudf::size_type inputRows = 75000;

  auto seq1 = cudf::test::make_counting_transform_iterator(0, [](auto i) { return i; });
  auto valid_seq1 = cudf::test::make_counting_transform_iterator(0, [](auto row) { return true; });
  cudf::test::fixed_width_column_wrapper<int64_t, typename decltype(seq1)::value_type> col1(seq1, seq1 + inputRows, valid_seq1);

  cudf::table_view input{{col1}};

  auto seq3 = cudf::test::make_counting_transform_iterator(0, [](auto i) {
    return (i % 277) == 0;
  });
  cudf::test::fixed_width_column_wrapper<bool> boolean_mask(seq3, seq3 + inputRows);

  auto got = cudf::apply_boolean_mask(input, boolean_mask);
  auto out_col = got->get_column(0).view();
  auto expected_null_count = cudf::count_unset_bits(out_col.null_mask(), 0, out_col.size());

  ASSERT_EQ(out_col.null_count(), expected_null_count);
}
```

**Expected behavior**
`cudf::apply_boolean_mask` should return a table containing columns with the correct null_count

### LINKED ISSUE #7210: [BUG] Test column comparison functions do not compare column null_count()
The comparison of `null_count()` inside of `column_property_comparator::compare_common()` between input columns seems to have been erroneously removed.   

Adding it back in causes a few unit tests to fail (1 in RESHAPE_TEST, 1 in STREAM_COMPACTION_TEST and a number of them in COPYING_TEST)

`EXPECT_EQ(lhs.null_count(), rhs.null_count());`

### PR DESCRIPTION
**Pull request by nvdbaranec**
Fixes #7210
Fixes #6733 

List of fixes included:

- [x] Restore `null_count()` check in `expect_columns_equal` / `expect_columns_equivalent`
- [x] Fix issue in `structs_column_view::get_sliced_child`
- [x] Fix test failures in COPYING_TEST
- [x] Fix test failures in STREAM_COMPACTION_TEST
- [x] Fix test failures in RESHAPE_TEST

### CODE REVIEW

File: cpp/src/structs/structs_column_view.cpp [hunk 1]
```diff
@@ -36,7 +36,9 @@ column_view structs_column_view::get_sliced_child(int index) const
                      size(),
                      child(index).head<uint8_t>(),
                      child(index).null_mask(),
-                     child(index).null_count(),
+                     // TODO: could potentially compute the actual count here, but at
+                     // the moment this interface doesn't take a stream.
+                     UNKNOWN_NULL_COUNT,
```
Comment (mythrocks) [inline cpp/src/structs/structs_column_view.cpp:41]: Would it be incorrect to lean into this and remove the `TODO`?

Comment (nvdbaranec) [reply cpp/src/structs/structs_column_view.cpp:41]: You mean change the interface to take a stream?  I think I'd be inclined to do that as a followup PR.

File: cpp/tests/utilities/column_utilities.cu [hunk 2]
```diff
@@ -65,8 +65,11 @@ struct column_property_comparator {
 
     EXPECT_EQ(lhs.size(), rhs.size());
 
+    EXPECT_EQ(lhs.null_count(), rhs.null_count());
```
Comment (harrism) [inline cpp/tests/utilities/column_utilities.cu:68]: Oops, we have a duplicate here...

Comment (harrism) [reply cpp/tests/utilities/column_utilities.cu:68]: @nvdbaranec shouldn't we only be checking for null_count equality if `check_exact_equality` is true, like this? 

```
    if (lhs.size() > 0 && check_exact_equality) {
      EXPECT_EQ(lhs.nullable(), rhs.nullable());
      EXPECT_EQ(lhs.null_count(), rhs.null_count());
    }
```

Comment (jrhemstad) [reply cpp/tests/utilities/column_utilities.cu:68]: I think null_count should still be checked even if just doing equivalence. There shouldn't be a situation where two equivalent columns have different null_counts.

Comment (harrism) [reply cpp/tests/utilities/column_utilities.cu:68]: Thanks Jake. Will just remove the duplicate line here then.

**Review by harrism — APPROVED** [review]

File: cpp/src/structs/structs_column_view.cpp (hunk 1 above)
Comment (mythrocks) [reply cpp/src/structs/structs_column_view.cpp:41]: Sorry, I meant that we remove the `TODO`, and leave it at `UNKNOWN_NULL_COUNT`.

On second thought, maybe we should leave the `TODO` in as a reminder to revisit.

**Review by mythrocks — APPROVED** [review]

**Review by jrhemstad — APPROVED** [review]

**Comment by harrism** [comment]
@gpucibot merge

**Pull request merged**